Draft of the project management

izabelcavassim opened the pull request:
here it is a draft of the possible topics to be included in the session

izabelcavassim committed:
started with the planning for the session

izabelcavassim committed:
updated

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+
+- [RStudio support](https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects)
+
+## Best practices
```
Might be good to list the best practices here, then go into more detail for each of them after that.

lwjohnst86 commented on intro-project_management.Rmd:
````diff
@@ -0,0 +1,235 @@
+library(ggplot2); library(dplyr)
+```
+
+### Indent your code
````
This would be better as "Use a syntax style guide" and then link to the tidyverse style guide: https://style.tidyverse.org/

izabelcavassim replied:
I am unable to install the package styler. if you can make it to work on their computer then fine!

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+
+## Creating your first project
+
+### Project
```
For this section, I would follow the format that R packages have:

- R/
- doc/
- data/
- data-raw/

I created a package ([prodigenr](http://prodigenr.lukewjohnston.com/)) to help set up this structure. I can add that part to material.

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+1. Put each project in its own directory, which is named after the project.
+2. Put text documents associated with the project in the **doc** directory.
+3. Put raw data and metadata in the **data** directory, and files generated during cleanup and analysis in a **results** directory.
+4. Put source for the project's scripts and programs in the **src** directory, and programs brought in from elsewhere or compiled locally in the **bin** directory.
```
The one about "bin" isn't really common or necessary for most R data analyses. 

Instead of "src", use "R/".

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+6. Name all files to reflect their content or function.
+
+
+### Exercise: Build and outline your first project
```
This section and exercise would be better done first. The exercise might not be necessary since we would do the code along making the project directory with them.

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+4. src
+5. bin
+
+### Advantages of using projects
```
Great section :)

lwjohnst86 replied:
This could also be moved up sooner.

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+
+- A new R session (process) is started
+- The .Rprofile file in the project's main directory (if any) is sourced by R
+- The .RData file in the project's main directory is loaded (if project options indicate that it should be loaded).
```
The `.RData` file should generally not be used. Its better to have a clean slate when starting your project, rather than starting with the environment cluttered.

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+- **Instructor**: `r params$instructor`
+    - **Contributions from**: Maria Izabel Cavassim Alves ([\@izabelcavassim](https://github.com/izabelcavassim))
+- **Session level**: `r params$level`
+    - **Part of the ["Beginner Series"](index.html)**
```
Part of the "Analysis workflow" series

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+- [Organizing R Source Code](https://www.r-bloggers.com/r-best-practices-r-you-writing-the-r-way/)
+- [An example of a well organised folder project](https://github.com/hadley/data-baby-names)
+- [How to create repetitive reports](https://learnr.wordpress.com/2009/09/09/brew-creating-repetitive-reports/)
+- [Best practices on writing functions in R](https://www.stat.cmu.edu/~cshalizi/402/programming/writing-functions.pdf)
```
Good list!

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+
+###  Describe your code
+
+It is very important for your future self, and for any person that will be reading/using your code to be able to understand what the **code**, **function** or **RMardown** will generate. Therefore it is cructial to describe what the code does, acknowledge the author, the date it was written,and give an example on how to execute it.
```
Here I would also add that the function name itself should be descriptive enough to say what the code does.

lwjohnst86 commented on intro-project_management.Rmd:
````diff
@@ -0,0 +1,235 @@
+  if(character_operation == "/"){results = a/b}
+return(results)
+}
+```
````
Maybe instead of this, copy the example "bad code" from the tidyverse style guide and get them to revise them to be more descriptive.

lwjohnst86 commented on intro-project_management.Rmd:
````diff
@@ -0,0 +1,235 @@
+```
+
+### Comment your code!
+Whenever we are dealing with many codes of lines, it is important to comment on what possible functions important lines of the code would have. You can use the symbol **#** to make comments:
````
Here again comment that first and foremost the code should be easy to read and understandable to others... So `x` to describe a vector of temperatures is not a good name to use, but `temperature_celsius_values` is better. Even though it is longer, it is better readability and maintainability.

lwjohnst86 commented on intro-project_management.Rmd:
````diff
@@ -0,0 +1,235 @@
+
+```{r}
+# This function converts temperature in fahrenheit to kelvin
+fahrenheit_to_kelvin <- function(temp_F) {
````
For instance, `temp_K` would be better written as `kelvin_temperature`.

izabelcavassim replied:
maybe i could let them to write this exercise again in the way they thing is the most appropriated once I have told them about those concepts.

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+
+### Organise all source files in the same directory
+
+It is a good idea to store all the necessary files that will be used/sourced in your code in the same directory. You can use the respective relative path to access them. You will learn later how to do it through R projects.
```
Really should strongly emphasize this. You *should* put all files associated with a project in one folder. A project is a product, a "package" or "bundle". A tiny machine. All parts of the bundle/machine need to be in one place.

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,235 @@
+
+### Data Cleaning
+
+In many cases your data will be "dirty": it will need significant preprocessing to get into a format R (or any other programming language) will find useful. This task is sometimes called "data munging". I find it useful to store these scripts in a separate folder, and create a second "read-only" data folder to hold the "cleaned" data sets. 
```
This should part about "read-only" should be emphasized more. This is so so so important!

lwjohnst86 replied:
Messy might be a better word than dirty. Dirty in English might invoke an image of bad or unusable or corrupt (e.g. "dirty politician" or a politician who is corrupt). Messy can be tidied.

lwjohnst86 reviewed (changes requested):
Great first draft!! I've made some comments. I'll come back to making more suggestions later in the afternoon (I have a PhD defense soon). In the meantime, you can revise as you can and take a look at a document I started writing [here](http://prodigenr.lukewjohnston.com/articles/manifesto.html). Also take a look at [rrtools](https://github.com/benmarwick/rrtools) (the references section of the README is great).

izabelcavassim committed:
made some changes, added some text. waiting for revision

izabelcavassim committed:
Delete .DS_Store

lwjohnst86 commented on intro-project_management.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+**For learning**:
+
+- [Good enough practicies in scientific computing ](https://github.com/swcarpentry/good-enough-practices-in-scientific-computing/blob/gh-pages/good-enough-practices-for-scientific-computing.pdf)
```
Better to use the DOI  https://doi.org/10.1371/journal.pcbi.1005510

lwjohnst86 replied:
Probably good to add the "Best practices" version by the same group  https://doi.org/10.1371/journal.pbio.1001745

lwjohnst86 reviewed (commented):
I don't have much time tomorrow to make more comments and also prepare, so I'll take over making edits and formatting for the website. I'll make comments describing what I changed, what I added, etc, so you can also learn! :) You get back from US in the summer of next year right? Hopefully you can teach by then! :)

izabelcavassim commented:
Thanks Luke, I am so sorry to put you in this position again.

I hope I will be able to teach by then! Thank you very much! This week i am
also submitting my paper! So it is a bit stressful!

Best,

Maria Izabel

On Mon, Dec 3, 2018 at 10:17 PM Luke W Johnston <notifications@github.com>
wrote:

> *@lwjohnst86* commented on this pull request.
>
> I don't have much time tomorrow to make more comments and also prepare, so
> I'll take over making edits and formatting for the website. I'll make
> comments describing what I changed, what I added, etc, so you can also
> learn! :) You get back from US in the summer of next year right? Hopefully
> you can teach by then! :)
> ------------------------------
>
> In intro-project_management.Rmd
> <https://github.com/au-oc/content/pull/13#discussion_r238435626>:
>
> > +
> +1. To use and learn the best practices for project organization.
> +1. To use RStudio to create and manage projects with consistent layout.
> +
> +**At the end of this session you will be able**:
> +
> +- To apply the best practices using R.
> +- To create a new Rstudio project.
> +- To organise folders in a consistent, structured and systematic way.
> +
> +
> +## Resources for learning and help
> +
> +**For learning**:
> +
> +- [Good enough practicies in scientific computing ](https://github.com/swcarpentry/good-enough-practices-in-scientific-computing/blob/gh-pages/good-enough-practices-for-scientific-computing.pdf)
>
> Better to use the DOI https://doi.org/10.1371/journal.pcbi.1005510
> ------------------------------
>
> In intro-project_management.Rmd
> <https://github.com/au-oc/content/pull/13#discussion_r238436230>:
>
> > +
> +1. To use and learn the best practices for project organization.
> +1. To use RStudio to create and manage projects with consistent layout.
> +
> +**At the end of this session you will be able**:
> +
> +- To apply the best practices using R.
> +- To create a new Rstudio project.
> +- To organise folders in a consistent, structured and systematic way.
> +
> +
> +## Resources for learning and help
> +
> +**For learning**:
> +
> +- [Good enough practicies in scientific computing ](https://github.com/swcarpentry/good-enough-practices-in-scientific-computing/blob/gh-pages/good-enough-practices-for-scientific-computing.pdf)
>
> Probably good to add the "Best practices" version by the same group
> https://doi.org/10.1371/journal.pbio.1001745
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/au-oc/content/pull/13#pullrequestreview-180977590>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHdmCcFo15Em40_jx2p91d--yl0brbsuks5u1ZTxgaJpZM4Y-ZHP>
> .
>


-- 
PhD in Bioinformatics at Bioinformatics Research Center (BiRC)
Aarhus University - Denmark
Phone:  +45 91458049
E-mail: mica@birc.au.dk
https://izabelcavassim.github.io/
Linkedin profile
<https://www.linkedin.com/profile/view?id=350503047&trk=nav_responsive_tab_profile>

lwjohnst86 commented:
I'll add comments about the changes tomorrow.

izabelcavassim commented:
Thanks!



On Tue, Dec 4, 2018 at 4:06 PM Luke W Johnston <notifications@github.com>
wrote:

> I'll add comments about the changes tomorrow.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/au-oc/content/pull/13#issuecomment-444132661>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHdmCQIACwwvFxVIvlYOPZFeedxAcBetks5u1o-SgaJpZM4Y-ZHP>
> .
>


-- 
PhD in Bioinformatics at Bioinformatics Research Center (BiRC)
Aarhus University - Denmark
Phone:  +45 91458049
E-mail: mica@birc.au.dk
https://izabelcavassim.github.io/
Linkedin profile
<https://www.linkedin.com/profile/view?id=350503047&trk=nav_responsive_tab_profile>

lwjohnst86 commented:
@izabelcavassim Not sure why the updated version isn't here, so I can't make direct line comments. So I'll write out a bit on what I changed and why, as well as some feedback for you to help with making more lessons in the future :)

- The biggest change I made was switch the project creation and setup to the beginning and expanded on this a bit more. I also introduced the prodigenr package to automate creating projects. I changed this because the folder and file setup is the first and probably simplest thing people can do to make their workflow and project management easier. It lays the foundation for later analyses.
- Next was to go over syntax styling and styler. Naming things is super important and is also a simple way to improve the workflow and setup. I changed the exercise to show examples of bad code and bad filenaming, and then let them figure out what was bad about the code/file names. 
- Other than that, I just reworded things to fit my voice (it's easier for me to prepare and teach material if I give it my voice. When you teach, the material will have your own voice too!) and made some formatting and small structural changes to fit the website and previous sessions.

You did a great job writing up a draft of this! Thanks! Hopefully you can get to teaching this next year some time!! :) Happy holidays!

izabelcavassim commented:
Hej Luke thanks for the constructive feedback, I am really sorry for not
being able to attend it!
I really like teaching and that would be a great opportunity to teach
people with a different background than bioinformatics.

Apart from that, since i didn't teach any time, i would like to help you in
writing the documentation for next year. I still have some days to spare
here in Denmark, so
let me know if you wanna discuss on the possible materials we could start
developing!

Thank you very much for the initiative, and I hope people are enjoying the
course.

Cheers,

Maria Izabel

On Wed, Dec 5, 2018 at 5:08 PM Luke W Johnston <notifications@github.com>
wrote:

> @izabelcavassim <https://github.com/izabelcavassim> Not sure why the
> updated version isn't here, so I can't make direct line comments. So I'll
> write out a bit on what I changed and why, as well as some feedback for you
> to help with making more lessons in the future :)
>
>    - The biggest change I made was switch the project creation and setup
>    to the beginning and expanded on this a bit more. I also introduced the
>    prodigenr package to automate creating projects. I changed this because the
>    folder and file setup is the first and probably simplest thing people can
>    do to make their workflow and project management easier. It lays the
>    foundation for later analyses.
>    - Next was to go over syntax styling and styler. Naming things is
>    super important and is also a simple way to improve the workflow and setup.
>    I changed the exercise to show examples of bad code and bad filenaming, and
>    then let them figure out what was bad about the code/file names.
>    - Other than that, I just reworded things to fit my voice (it's easier
>    for me to prepare and teach material if I give it my voice. When you teach,
>    the material will have your own voice too!) and made some formatting and
>    small structural changes to fit the website and previous sessions.
>
> You did a great job writing up a draft of this! Thanks! Hopefully you can
> get to teaching this next year some time!! :) Happy holidays!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/au-oc/content/pull/13#issuecomment-444541385>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHdmCRcxhlNRMkTHi8SXJu7WeWZnP3DJks5u1--QgaJpZM4Y-ZHP>
> .
>


-- 
PhD in Bioinformatics at Bioinformatics Research Center (BiRC)
Aarhus University - Denmark
Phone:  +45 91458049
E-mail: mica@birc.au.dk
https://izabelcavassim.github.io/
Linkedin profile
<https://www.linkedin.com/profile/view?id=350503047&trk=nav_responsive_tab_profile>
